File: app/src/main/java/net/wojteksz128/worktimemeasureapp/notification/endOfWork/EndOfWorkNotificationIntentService.java
package net.wojteksz128.worktimemeasureapp.notification.endOfWork;

import android.app.IntentService;
import android.content.Intent;
import android.support.annotation.Nullable;

public class EndOfWorkNotificationIntentService extends IntentService {

    /**
     * Creates an IntentService.  Invoked by your subclass's constructor.
     */
    public EndOfWorkNotificationIntentService() {
        super(EndOfWorkNotificationIntentService.class.getName());
    }

    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        assert intent != null;
        final String action = intent.getAction();
        EndOfWorkNotification.Action.valueOf(action).doAction(this);
    }
}

File: app/src/main/kotlin/net/wojteksz128/worktimemeasureapp/notification/endOfWork/EndOfWorkNotificationIntentService.kt
package net.wojteksz128.worktimemeasureapp.notification.endOfWork

import android.app.IntentService
import android.content.Intent
import net.wojteksz128.worktimemeasureapp.notification.action.Action

/**
 * Creates an IntentService. Invoked by your subclass's constructor.
 */
class EndOfWorkNotificationIntentService : IntentService(EndOfWorkNotificationIntentService::class.java.name) {

    override fun onHandleIntent(intent: Intent) {
        val action = intent.action
        action?.let { Action.valueOf(it)(this) }
    }
}
